File: benchmarks/java/set/Clojure_Set_From_Java/TrafficLightExec.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class TrafficLightExec {



    private TrafficLight _TrafficLight = new TrafficLight();



    private BInteger counter;

    public TrafficLightExec() {
        counter = new BInteger(0);
    }

    public void simulate() {
        while((counter.less(new BInteger(500000))).booleanValue()) {
            this._TrafficLight.cars_ry();
            this._TrafficLight.cars_g();
            this._TrafficLight.cars_y();
            this._TrafficLight.cars_r();
            this._TrafficLight.peds_g();
            this._TrafficLight.peds_r();
            counter = counter.plus(new BInteger(1));
        }
    }

    public static void main(String[] args) {
        TrafficLightExec exec = new TrafficLightExec();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));

        Runtime runtime = Runtime.getRuntime();
        long memory = runtime.totalMemory() - runtime.freeMemory();
        System.out.println("Used memory is bytes: " + memory);
        System.out.println("Used memory is megabytes: "
                + (memory / (1024L * 1024L)));
    }

}

File: benchmarks/C++_Primitives/TrafficLightExec.cpp
#include <iostream>
#include <string>
#include "BUtils.h"
#include "BInteger.h"
#include "BBoolean.h"
#include "TrafficLight.cpp"

#ifndef TrafficLightExec_H
#define TrafficLightExec_H

using namespace std;

class TrafficLightExec {

    public:

    private:


        TrafficLight _TrafficLight;


        BInteger counter;

    public:

        TrafficLightExec() {
            counter = (BInteger(0));
        }

        void simulate() {
            while((counter.less((BInteger(300000000)))).booleanValue()) {
                this->_TrafficLight.cars_ry();
                this->_TrafficLight.cars_g();
                this->_TrafficLight.cars_y();
                this->_TrafficLight.cars_r();
                this->_TrafficLight.peds_g();
                this->_TrafficLight.peds_r();
                counter = counter.plus((BInteger(1)));
            }
        }
};

int main() {
    TrafficLightExec exec;
    exec.simulate();
    return 0;
}

#endif